modules/programs/rio.nix

```
@@ -0,0 +1,51 @@
+{ lib, pkgs, config, ... }:
+let
+  cfg = config.programs.rio;
+
+  settingsFormat = pkgs.formats.toml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+in {
+  options.programs.rio = {
+    enable = lib.mkEnableOption null // {
+      description = lib.mdDoc ''
+        Enable Rio, a terminal built to run everywhere, as a native desktop applications by
+        Rust/WebGPU or even in the browsers powered by WebAssembly/WebGPU.
+      '';
+    };
+
+    package = lib.mkPackageOption pkgs "rio" { };
+
+    settings = lib.mkOption {
+      type = lib.types.either lib.types.path settingsFormat.type;
```

The settings option is intended only for structured configurations

```suggestion
      type = settingsFormat.type;
```

---

If you want to allow the user to add arbitrary configuration, then you can add an `extraConfig` option of type `lines` that can be merged into the configuration file. But if the format really is plain toml then there should be no need, I believe the Nix syntax covers all toml features.